servers/services/src/test/java/org/projectnessie/services/impl/AbstractTestDiff.java

```diff
@@ -80,48 +80,20 @@ public void diffPaging(int numKeys) throws BaseNessieClientServerException {
     int pageSize = 5;
 
     if (numKeys > 0) {
-      CommitMultipleOperationsBuilder commit =
-          getApi()
-              .commitMultipleOperations()
-              .branch(branch)
-              .commitMeta(CommitMeta.fromMessage("commit"));
-      for (int i = 0; i < numKeys; i++) {
-        commit.operation(Put.of(contentKey.apply(i), table.apply(i)));
-      }
-      branch = commit.commit();
+      branch =
+          commit(
+                  branch,
+                  fromMessage("commit"),
+                  IntStream.range(0, numKeys)
+                      .mapToObj(i -> Put.of(contentKey.apply(i), table.apply(i)))
+                      .toArray(Operation[]::new))
+              .getTargetBranch();
     }
 
-    Set<ContentKey> contents = new HashSet<>();
-    String token = null;
-    for (int i = 0; ; i += pageSize) {
-      DiffResponse response =
-          getApi()
-              .getDiff()
-              .fromRef(branch)
-              .toRef(defaultBranch)
-              .maxRecords(pageSize)
-              .pageToken(token)
-              .get();
-
-      for (DiffResponse.DiffEntry entry : response.getDiffs()) {
-        soft.assertThat(contents.add(entry.getKey()))
-            .describedAs("offset: %d , entry: %s", i, entry)
-            .isTrue();
-      }
-      soft.assertThat(contents).hasSize(Math.min(i + pageSize, numKeys));
-      if (i + pageSize < numKeys) {
-        soft.assertThat(response.getToken())
-            .describedAs("offset: %d", i)
-            .isNotEmpty()
-            .isNotEqualTo(token);
-        soft.assertThat(response.isHasMore()).describedAs("offset: %d", i).isTrue();
-        token = response.getToken();
-      } else {
-        soft.assertThat(response.getToken()).describedAs("offset: %d", i).isNull();
-        soft.assertThat(response.isHasMore()).describedAs("offset: %d", i).isFalse();
-        break;
-      }
-    }
+    Set<ContentKey> contents =
+        pagedDiff(branch, defaultBranch, pageSize, numKeys).stream()
+            .map(DiffEntry::getKey)
+            .collect(toSet());
 
     soft.assertThat(contents)
```

Comment:
Is this ever run with the current codebase? Would it make sense to use the page size == expected total size until diff paging is supported? Otherwise, this test code is effectively dead, it seems :thinking:

Reply:
Not with the current code base, but with #5641